lib/application/view-models/execution/collection_execution_vm.dart

```diff
@@ -3,70 +3,165 @@ import 'package:flutter_riverpod/flutter_riverpod.dart';
+  /// This action of marking will always trigger a new [LoadedCollectionExecutionState] update with the selected
+  /// [difficulty]
+  ///
+  /// Throws an [InconsistentStateError] if the state is other than [LoadedCollectionExecutionState]
+  void markCurrentMemoDifficulty(MemoDifficulty difficulty);
+
+  /// Go forward with the next suitable contents
+  ///
+  /// There are two scenarios where we should allow an execution to go forward:
+  ///   1. there is a question being displayed and it should provide the question to the current memo; or when
+  ///   2. there is an answer *with a marked answer* and it should proceed to next question (or change its state to a
+  /// [FinishedCollectionExecutionState] if there are no more memos left). In this case, if `nextContents` is requested
+  /// displaying an answer with no marked answer, it won't do anything.
+  ///
+  /// Throws an [InconsistentStateError] if the state is other than [LoadedCollectionExecutionState]
+  Future<void> nextContents();
 }
 
 class CollectionExecutionVMImpl extends CollectionExecutionVM {
-  CollectionExecutionVMImpl(this._services, {required this.collectionId}) : super(LoadingCollectionExecutionState()) {
+  CollectionExecutionVMImpl({
+    required this.executionServices,
+    required this.collectionServices,
+    required this.collectionId,
+  }) : super(LoadingCollectionExecutionState()) {
     _loadMemos();
   }
 
   final String collectionId;
-  final ExecutionServices _services;
+  final ExecutionServices executionServices;
+  final CollectionServices collectionServices;
 
   late final List<Memo> _memos;
+  late final Collection _collection;
   final _executions = <MemoExecution>[];
   var _currentMemoStartDate = DateTime.now().toUtc();
 
   bool get _isFinished => _memos.length == _executions.length;
 
   @override
-  Future<void> markCurrentMemoDifficulty(MemoDifficulty difficulty) async {
+  void markCurrentMemoDifficulty(MemoDifficulty difficulty) {
     if (state is! LoadedCollectionExecutionState) {
       throw InconsistentStateError.viewModel('Cannot mark the current memo before finishing loading');
     }
 
+    state = (state as LoadedCollectionExecutionState).copyWith(markedAnswer: difficulty);
+  }
+
+  @override
+  Future<void> nextContents() async {
+    if (state is! LoadedCollectionExecutionState) {
+      throw InconsistentStateError.viewModel('Cannot request next memo contents before finishing loading');
+    }
+
     final loadedState = state as LoadedCollectionExecutionState;
 
+    // If there is a `markedAnswer`, we assume that this is a request to forward with the last marked difficulty
+    if (loadedState.markedAnswer != null) {
+      await _confirmMarkedMemo(loadedState.markedAnswer!);
+    } else {
+      // Otherwise, we expect it to be a request to display the question
+      final currentMemo = _memos[_executions.length];
+      state = loadedState.copyWith(isDisplayingQuestion: false, currentContents: currentMemo.rawAnswer);
```

Comment:
The doc says that this is a request to display the question, but the state copy is displaying the answer

Reply:
I've updated the docs, it should always skip to the next suitable set of contents, meaning an answer or a question, no matter which order, the vm knows how to handle